operators/pkg/tenant-controller/sandbox_test.go

```diff
@@ -0,0 +1,253 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package tenant_controller_test
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	clv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+	clctx "github.com/netgroup-polito/CrownLabs/operators/pkg/context"
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/forge"
+	controllers "github.com/netgroup-polito/CrownLabs/operators/pkg/tenant-controller"
+	. "github.com/netgroup-polito/CrownLabs/operators/pkg/utils/tests"
+)
+
+var _ = Describe("Sandbox", func() {
+	var (
+		ctx           context.Context
+		clientBuilder fake.ClientBuilder
+		reconciler    controllers.TenantReconciler
+
+		tenant clv1alpha2.Tenant
+
+		sandboxNSname types.NamespacedName
+		sbNamespace   corev1.Namespace
+
+		roleBindingNSname types.NamespacedName
+		sbRoleBinding     rbacv1.RoleBinding
+		roleRef           rbacv1.RoleRef
+		subjects          []rbacv1.Subject
+
+		resQuotaNSname types.NamespacedName
+		sbResQuota     corev1.ResourceQuota
+
+		emptyStatus    clv1alpha2.NameCreated
+		presenceStatus clv1alpha2.NameCreated
+		ownerRef       metav1.OwnerReference
+		err            error
+	)
+
+	const (
+		tenantName = "tester"
+	)
+	InitializeTestStructs := func() {
+		tenant = clv1alpha2.Tenant{
+			ObjectMeta: metav1.ObjectMeta{Name: tenantName},
+		}
+		ownerRef = metav1.OwnerReference{
+			APIVersion:         clv1alpha2.GroupVersion.String(),
+			Kind:               "Tenant",
+			Name:               tenant.Name,
+			UID:                tenant.GetUID(),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+			Controller:         pointer.BoolPtr(true),
+		}
+
+		emptyStatus = clv1alpha2.NameCreated{
+			Name:    "",
+			Created: false,
+		}
+		presenceStatus = clv1alpha2.NameCreated{
+			Name:    sandboxNSname.Name,
+			Created: true,
+		}
+		sandboxNSname = types.NamespacedName{
+			Name: forge.CanonicalSandboxName(tenantName),
+		}
+		roleBindingNSname = types.NamespacedName{
+			Name:      "sandbox-editor",
+			Namespace: sandboxNSname.Name,
+		}
+		resQuotaNSname = types.NamespacedName{
+			Name:      "sandbox-resource-quota",
+			Namespace: sandboxNSname.Name,
+		}
+		sbNamespace = corev1.Namespace{}
+		sbRoleBinding = rbacv1.RoleBinding{}
+		sbResQuota = corev1.ResourceQuota{}
+
+		roleRef = rbacv1.RoleRef{Kind: "ClusterRole", Name: "edit", APIGroup: rbacv1.GroupName}
+		subjects = []rbacv1.Subject{{Kind: rbacv1.UserKind, Name: tenant.Name, APIGroup: rbacv1.GroupName}}
```

Comment:
Do not define as variables objects that are used only once. Just compare them inline. That makes the tests easier to understand, as you immediately see what you are asserting.